systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java

```diff
@@ -0,0 +1,334 @@
+                            .withNewKafkaListenerAuthenticationScramSha512Auth()
+                            .endKafkaListenerAuthenticationScramSha512Auth()
+                            .withNetworkPolicyPeers(
+                                new NetworkPolicyPeerBuilder()
+                                    .withNewPodSelector()
+                                    .withMatchLabels(matchLabelsForTls)
+                                    .endPodSelector()
+                                    .build())
+                        .endGenericKafkaListener()
+                    .endListeners()
+                .endKafka()
+            .endSpec()
+            .done();
+
+        String topic0 = "topic-example-0";
+        String topic1 = "topic-example-1";
+        KafkaTopicResource.topic(CLUSTER_NAME, topic0).done();
+        KafkaTopicResource.topic(CLUSTER_NAME, topic1).done();
+
+        String userName = "user-example";
+        KafkaUser kafkaUser = KafkaUserResource.scramShaUser(CLUSTER_NAME, userName).done();
+
+        KafkaClientsResource.deployKafkaClients(true, allowedKafkaClientsName, kafkaUser).done();
+
+        String allowedKafkaClientsPodName = kubeClient().listPodsByPrefixInName(allowedKafkaClientsName).get(0).getMetadata().getName();
+
+        LOGGER.info("Verifying that {} pod is able to exchange messages", allowedKafkaClientsPodName);
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(allowedKafkaClientsPodName)
+            .withTopicName(topic0)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .withKafkaUsername(userName)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        internalKafkaClient.checkProducedAndConsumedMessages(
+            internalKafkaClient.sendMessagesTls(),
+            internalKafkaClient.receiveMessagesTls()
+        );
+
+        KafkaClientsResource.deployKafkaClients(true, deniedKafkaClientsName, kafkaUser).done();
+
+        String deniedKafkaClientsPodName = kubeClient().listPodsByPrefixInName(deniedKafkaClientsName).get(0).getMetadata().getName();
+
+        InternalKafkaClient newInternalKafkaClient = internalKafkaClient.toBuilder()
+            .withUsingPodName(deniedKafkaClientsPodName)
+            .withTopicName(topic1)
+            .withConsumerGroupName(ClientUtils.generateRandomConsumerGroup())
+            .build();
+
+        LOGGER.info("Verifying that {} pod is  not able to exchange messages", deniedKafkaClientsPodName);
+
+        assertThrows(AssertionError.class, () -> {
+            newInternalKafkaClient.checkProducedAndConsumedMessages(
+                newInternalKafkaClient.sendMessagesTls(),
+                newInternalKafkaClient.receiveMessagesTls()
+            );
+        });
+    }
+
+    @Test
+    void testNPWhenOperatorIsInSameNamespaceAsOperand() {
```

Comment:
Correct me if I am wrong, but you should ut there an assertion, that the install type is `bundle`. Same for the second test. Otherwise, it will be executed in other install types as well.

Reply:
The whole ST is "tagged" with `NETWORKPOLICIES_SUPPORTED` and these tests are not run for Olm and Helm. But maybe it will be better to add some other check.

Or ... @scholzj do we want to run these tests for Helm and OLM too?

Reply:
But you should add this assumption as well because it will install CO via yaml bundle always.